File: twoSumClosest.py
class Solution:

    def twoSumClosest(self, s, d):
        # write your code here
        s.sort()      
        i = 0
        j = len(s) - 1
        min = d - 30
        res = []
        largest = 0
        while(i < j):
            if(s[i] + s[j] <= d - 30):
                if(d - 30 - s[i] - s[j] < min):
                    min = d - 30 - s[i] - s[j]
                    res = [s[i], s[j]]
                    largest = max(s[i], s[j])
                elif(d - 30 - s[i] - s[j] == min):
                    if(s[i] > largest or s[j] > largest):
                        min = d - 30 - s[i] - s[j]
                        res = [s[i], s[j]]
                        largest = max(s[i], s[j])
                i += 1
            else:
                j -= 1
        
        return res


if __name__ == '__main__':

    movie_duration = [90, 85, 75, 60, 120, 150, 125]
    d = 250

    S = Solution()
    print(Solution.twoSumClosest(S, movie_duration, d))

File: twoSumClosest.java
public class Solution {
    /**
     * @param nums an integer array
     * @param target an integer
     * @return the difference between the sum and the target
     */
    public int twoSumClosest(int[] nums, int target) {
        // Write your code here
        
        Arrays.sort(nums);
        
        int min = target;
        
        int left = 0;
        int right = nums.length - 1;
        int largest = 0;
        int[] res = {};

        while (left < right) {
            if (nums[left] + nums[right] < target) {
                min = Math.min(min, target - nums[left] - nums[right]);
                left++;
            }
            else {
                min = Math.min(min, nums[left] + nums[right] - target);
                right--;
            }
        }
        
        return min;
    }

    public int main() {

        int [] movie_duration = {90, 85, 75, 60, 120, 150, 125};
        int d = 250;
        this.twoSumClosest(movie_duration, d);
    }
}